meta-balena-common/recipes-core/busybox/busybox_%.bbappend

```diff
@@ -7,3 +7,8 @@ SRC_URI:append = " \
 RDEPENDS:${PN}:append = " openssl"
 
 ALTERNATIVE_PRIORITY[hwclock] = "100"
+
+# glibc enables 64bit time for stat and similar routines only if both _FILE_OFFSET_BITS=64 and _TIME_BITS=64 are set on 32bit platforms.
+#
+#   See https://www.phoronix.com/scan.php?page=news_item&px=Glibc-More-Y2038-Work
+CFLAGS:append:class-target = " -DHAVE_PROC_UPTIME -D_TIME_BITS=64 -D_FILE_OFFSET_BITS=64"
```

Comment:
@acostach wondering whether it's reasonable to try and build gcc with these as default instead of having to add it to individual user space packages. It would save us from future errors.

Reply:
@alexgg actually glibc won't build with these macros on, it's the tools that get linked against glibc 2.34 that have to set them if they want glibc to use the 64bit time structures internally. Also, not all programs work with TIME_BITS=64, there were some that needed to either have this disabled, or looked into and investigated and made to build fine and work, which wasn't a simple task.

Reply:
thanks @acostach, didn't realize it was glibc configuration. Although now that you mention it I remember I may have asked this too in a previous review, not sure why it didn't stick :)